### DISCUSSION TITLE
updateTriggers not working

### DISCUSSION THREAD

**Discussion by iPOiZN** [visgl/deck.gl#9033] [Q&A] [answered]
I am encountering an issue with deck.gl in a Nuxt 3 project with `mapboxOverlay`. Every time I call the `render` function, deck.gl reloads the data instead of just updating `getFillColor`. This behavior prevents the `updateTriggers` from working as expected. Below is the relevant part of my code:

` const selectedIds = ref([]);

	function render(selected: any) {
		const layersArr = new H3HexagonLayer({
			id: 'h3-hexagon-layer',
			data: getFirstData(2, undefined, undefined),

			coverage: 0.9,
			autoHighlight: true,
			extruded: true,
			pickable: true,
			highlightColor: [255, 200, 54, 200],
			elevationRange: [0, 1200],
			getHexagon: (d) => {
				return d.hex_id;
			},
			getElevation: (d) => +d.revenue_prediction,

			getFillColor: (d) => {
				if (selectedIds.value.includes(d.hex_id)) {
					return [0, 0, 0];
				}
				const count = +d.revenue_prediction;
				const maxCount = 1200;

				const factor = 1 - count / maxCount;
				const lightFactor = factor * 0.7 + 1.2; // Увеличиваем светлость

				const color = new Uint8Array([2, 136, 253].map((c) => Math.floor(c * lightFactor)));

				return color;
			},
			transitions: {
				getPosition: 600,
				getFillColor: {
					duration: 300,
					enter: ([r, g, b]) => [r, g, b, 0], // fade in
				},
				radiusScale: {
					type: 'spring',
					stiffness: 0.01,
					damping: 0.15,
				},
			},
			getLineColor: (d) => [0, 255, 0, 128],
			onDataLoad: (d) => {
				console.log('ol');
			},

			onClick: (pickingInfo, event) => {
				if (!selectedIds.value.includes(pickingInfo.object.hex_id)) {
					selectedIds.value.push(pickingInfo.object.hex_id);
				} else {
					selectedIds.value.splice(selectedIds.value.indexOf(pickingInfo.object.hex_id), 1);
				}
				render(selectedIds.value);
			},

			updateTriggers: {
				getFillColor: [selected],
			},
			dataComparator: (oldData, newData) => {
				return isEqual(oldData, newData);
			},
		});

		deckOverlay.setProps({ layers: [layersArr] });
	}`

**Comment by Pessimistress** [comment, marked as answer]
You are supplying a new object to `data` on every render. Cache your data if it doesn't change.

See https://deck.gl/docs/developer-guide/performance#avoid-unnecessary-shallow-change-in-data-prop

**Reply by iPOiZN** [reply]
Good. The first problem is solved.
I am having another trouble with updating the properties of a layer, such as changing coverage, extruded, elevationScale, etc. In previous versions of deck.gl, these could be changed using the setProps function, but it no longer works. How can these properties be updated after the layer and data have been loaded?
